### PR TITLE
[TECH-525] Junit test cases

### PR DESCRIPTION
**Pull request by PoornaPanda9785**

### CODE REVIEW

File: src/test/java/com/hotel/mappers/book/response/AddressMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    @InjectMocks
+    private AddressMapper addressMapper;
+
+    @Test
+    public void map() {
+        AddressInfoType addressInfoType = getAddress();
+        Address address = addressMapper.map(addressInfoType);
+        assertThat(address).isNotNull();
+        assertThat(address.getStreetAddress()).isEqualTo("Hotel Paris, Street1, France");
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/book/response/AddressMapperTest.java:24]: add assertThat() for countryName also

File: src/test/java/com/hotel/mappers/book/response/PnrInfoMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.opentravel.ota._2003._05.StatusMode;
+import org.opentravel.ota._2003._05.TPAExtensionsType;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+public class PnrInfoMapperTest {
+
+          @InjectMocks
+          PnrInfoMapper pnrInfoMapper;
+
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/book/response/PnrInfoMapperTest.java:19]: Reformat code

File: src/test/java/com/hotel/mappers/book/response/RateMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opentravel.ota._2003._05.TaxesType;
+import org.opentravel.ota._2003._05.TotalType;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/book/response/RateMapperTest.java:17]: Remove import *

File: src/test/java/com/hotel/mappers/book/response/RateMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@ExtendWith(MockitoExtension.class)
+public class RateMapperTest {
+
+    @InjectMocks
+    RateMapper rateMapper;
+
+    @Test
+ public void map() {
+        TotalType rateResponse = getRate();
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/book/response/RateMapperTest.java:26]: Reformat code

File: src/test/java/com/hotel/mappers/common/POSMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import com.hotel.service.common.Context;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opentravel.ota._2003._05.ArrayOfSourceType;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class POSMapperTest {
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/common/POSMapperTest.java:12]: add public for class

File: src/test/java/com/hotel/mappers/common/POSMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    @InjectMocks
+    public POSMapper posMapper;
+
+    @Test
+    void mapPOS() {
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/common/POSMapperTest.java:17]: Access modifier, public is missing

File: src/test/java/com/hotel/service/book/BookServerServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class BookServerServiceTest {
+
+    @Mock
+    private DjocaClient djocaClient;
+
+    @InjectMocks
+  private BookServerService bookServerService;
+
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/service/book/BookServerServiceTest.java:33]: Reformat code

File: src/test/java/com/hotel/service/book/BookServerServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        HotelBookRequest request = getRequest();
+        OTAHotelResRS otaHotelResRS = new OTAHotelResRS();
+        OTAHotelResRQ otaHotelResRQ = new OTAHotelResRQ();
+        HotelBookResponse.Builder hotelBookResponse = HotelBookResponse.newBuilder();
+        StreamRecorder<HotelBookResponse> responseStreamRecorder = StreamRecorder.create();
+        when(hotelReservationsMapper.map(request)).thenReturn(otaHotelResRQ);
+        when(djocaClient.restClient(otaHotelResRQ, request.getRequestContext().getSupplierUrl(), BOOK_SERVICE)).thenReturn(otaHotelResRS);
+        when(hotelBookResponseMapper.map(otaHotelResRS)).thenReturn(hotelBookResponse.build());
+        bookServerService.getHotelBook(request,responseStreamRecorder);
+        verify(hotelReservationsMapper, atLeast(1)).map(request);
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/service/book/BookServerServiceTest.java:51]: add assertThat on the response getting from responseStreamRecorder

File: src/test/java/com/hotel/service/raterule/HotelRateRuleServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(MockitoExtension.class)
+public class HotelRateRuleServiceTest {
+
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/service/raterule/HotelRateRuleServiceTest.java:27]: The test class for HotelRateRuleService might added by Team 1. Please check with them and remove if they already added

File: src/test/java/com/hotel/mappers/book/request/GuaranteeMapperTest.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        GuaranteeType guarantee = guaranteeMapper.map(paymentInfo);
+        assertThat(guarantee).isNotNull();
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getGuaranteeTypeCode()).isEqualTo("Card");
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getPaymentCard().getCardCode()).isEqualTo("Visa");
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getPaymentCard().getCardNumber()).isEqualTo("1234");
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getPaymentCard().getSeriesCode()).isEqualTo("123");
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getPaymentCard().getExpireDate()).isEqualTo("2023-11-18");
+        assertThat(guarantee.getGuaranteesAccepted().getGuaranteeAccepted().get(0).getPaymentCard().getCardHolderName()).isEqualTo("Test");
+
+
```
Comment (LubaibaZ) [inline src/test/java/com/hotel/mappers/book/request/GuaranteeMapperTest.java:39]: Remove extra space

**Review by LubaibaZ — CHANGES_REQUESTED** [review]

File: src/test/java/com/hotel/mappers/book/request/CustomerTypeMapperTest.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        assertThat(customer).isNotNull();
+        assertThat(customer.getPersonName().getNamePrefix().get(0)).isEqualTo("Mr");
+        assertThat(customer.getPersonName().getGivenName().get(0)).isEqualTo("Test");
+        assertThat(customer.getTelephone().get(0).getPhoneNumber()).isEqualTo("1234");
+        assertThat(customer.getEmail().get(0).getValue()).isEqualTo("test@gmail.com");
+        assertThat(customer.getAddress().get(0).getCityName()).isEqualTo("Paris");
+        assertThat(customer.getAddress().get(0).getCountryName().getCode()).isEqualTo("FR");
+        assertThat(customer.getAddress().get(0).getCountryName().getValue()).isEqualTo("France");
+       verify(customerAddressMapper,atLeast(1)).mapAddress(userInfo.getAddress());
+
```
[the remark this answers is not in the document]
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/book/request/CustomerTypeMapperTest.java]: Done

File: src/test/java/com/hotel/mappers/book/response/AddressMapperTest.java (hunk 1 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/book/response/AddressMapperTest.java:24]: Done added City name too

File: src/test/java/com/hotel/mappers/book/response/PnrInfoMapperTest.java (hunk 2 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/book/response/PnrInfoMapperTest.java:20]: Done

File: src/test/java/com/hotel/mappers/book/response/RateMapperTest.java (hunk 3 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/book/response/RateMapperTest.java:16]: Done

File: src/test/java/com/hotel/mappers/book/response/RateMapperTest.java (hunk 4 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/book/response/RateMapperTest.java:25]: Done

File: src/test/java/com/hotel/mappers/common/POSMapperTest.java (hunk 6 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/mappers/common/POSMapperTest.java]: Corrected that

File: src/test/java/com/hotel/service/raterule/HotelRateRuleServiceTest.java (hunk 9 above)
Comment (PoornaPanda9785) [reply src/test/java/com/hotel/service/raterule/HotelRateRuleServiceTest.java:27]: Removed it

**Review by LubaibaZ — APPROVED** [review]

**Pull request merged**